src/modules/scraper/adapter/messaging/DepositReferralConsumer.ts

```diff
@@ -1,19 +1,57 @@
 import { OnQueueFailed, Process, Processor } from "@nestjs/bull";
 import { Logger } from "@nestjs/common";
 import { Job } from "bull";
-import { DepositReferralQueueMessage, ScraperQueue } from ".";
+import { Repository } from "typeorm";
+import { InjectRepository } from "@nestjs/typeorm";
+
+import { DepositReferralQueueMessage, RectifyStickyReferralQueueMessage, ScraperQueue } from ".";
 import { ReferralService } from "../../../referral/services/service";
+import { Deposit } from "../../../deposit/model/deposit.entity";
+import { EthProvidersService } from "../../../web3/services/EthProvidersService";
+import { ScraperQueuesService } from "../../service/ScraperQueuesService";
 
 @Processor(ScraperQueue.DepositReferral)
 export class DepositReferralConsumer {
   private logger = new Logger(DepositReferralConsumer.name);
 
-  constructor(private referralService: ReferralService) {}
+  constructor(
+    private referralService: ReferralService,
+    @InjectRepository(Deposit) readonly depositRepository: Repository<Deposit>,
+    private ethProvidersService: EthProvidersService,
+    private scraperQueuesService: ScraperQueuesService,
+  ) {}
 
   @Process({ concurrency: 1 })
   private async process(job: Job<DepositReferralQueueMessage>) {
-    const { depositId } = job.data;
-    await this.referralService.extractReferralAddressAndComputeStickyReferralAddresses(depositId);
+    const { depositId, rectifyStickyReferralAddress } = job.data;
+    const deposit = await this.depositRepository.findOne({ where: { id: depositId } });
+
+    if (!deposit) return;
+    // if depositDate field is missing, throw an error to retry the message as this field
+    // is necessary to compute the sticky referral address.
+    if (!deposit.depositDate) throw new Error(`depositId ${deposit.id}: wait for depositDate`);
+
+    const { depositTxHash, sourceChainId } = deposit;
+    const transaction = await this.ethProvidersService.getCachedTransaction(sourceChainId, depositTxHash);
+    const block = await this.ethProvidersService.getCachedBlock(sourceChainId, transaction.blockNumber);
+    const blockTimestamp = parseInt((new Date(block.date).getTime() / 1000).toFixed(0));
+
+    if (!transaction) throw new Error("Transaction not found");
```

Comment:
Is retrying the message in this case the wanted behavior? If a transaction is not found then
1. Either the tx wasn't indexed or mined
2. Or a tx with the hash doesn't exist

Case 1 can be excluded because the deposit was stored through an emitted contract event. And if case 2, then this should just return, no?

Reply:
Yes, retrying is the wanted behaviour, because, as you said, this is very unlikely to happen, but if it happens, then it means that there's definitely a problem somewhere. So I prefer the message to be retried, so when the issue/bug is fixed, then the message will be retried successfully